### PR TITLE
BACKEND: Adds 2020 territory population data pipeline

### LINKED ISSUE #1954: Find missing territory county-equivalent populations
## Is your feature request related to a problem? Please describe.

Right now if you view St John (a county equivalent of U.S. Virgin Islands), there is no population listed in the population card. I believe these populations are available in the same 2010 tables we used manually to get the territory state-level populations per demographic. 

## Describe the solution you'd like

Manually update the `/data` tables containing 2010 ACS with missing county-equivalent population totals. Is there newer territory population data??

### LINKED ISSUE #2032: Load Territory Populations from 2020 Island Areas Census
## Is your feature request related to a problem? Please describe.

our territory population is from 2010 for the smaller territories

## Describe the solution you'd like

2020 data is available here: 

USVI: https://data.census.gov/table?g=0400000US78&d=DECIA+U.S.+Virgin+Islands+Demographic+Profile&tid=DECENNIALDPVI2020.DP1

- St John: https://data.census.gov/table?q=population+by+race+in+St.+John+Island,+United+States+Virgin+Islands&tid=DECENNIALDPVI2020.DP1&tp=false

Guam: https://data.census.gov/table?q=population+by+race+in+guam&tid=DECENNIALDPGU2020.DP1

American Samoa: https://data.census.gov/table?q=population+by+race+in+american+samoa&tid=DECENNIALDPAS2020.DP1

Northern Mariana Islands: https://data.census.gov/table?q=population+by+race+in+Commonwealth+of+the+Northern+Mariana+Islands&tid=DECENNIALDPMP2020.DP1

## Which of our [user persona(s)](https://docs.google.com/document/d/1EASpK_THTE_uy_Yk0sut2GTVd3w5pKtKH7LpLtF-0co/) would this feature primarily benefit?

- Informer: helps everyday people understand important current events (e.g. journalists working in storytelling settings)
- Persuader: passage of legislation by presenting concise and compelling arguments (e.g. staff members, policy advisors in legislative or nonprofit settings)

### PR DESCRIPTION
**Pull request by benhammondmusic**
## Description

- Adds a new data pipeline to ingest downloaded 2020 decennial tables for the smaller territories not shown in the ACS 5-year estimates
- write into HET style tables on BigQuery for both STATE and COUNTY equivalent tables
- all 4 islands get age/sex/race data; USVI only also gets NH races
- adds util fn for melting group/metric col data into HET-style row-based group, col-based metric data.

TODO: Once those are in place, we can adjust our `acs_population` to swap its ingestion for those territories from `_2010` to `_2020`. 

LINKS: 
- [USVI state + counties](https://data.census.gov/table?g=0400000US78,78$0500000&d=DECIA+U.S.+Virgin+Islands+Demographic+Profile)
- [Guam state + counties](https://data.census.gov/table?q=population+by+race+in+guam&g=0400000US66$0500000)
- [American Samoa state + counties](https://data.census.gov/table?q=population+by+race+in+american+samoa&g=0400000US60$0500000)
- [Mariana Islands state + counties](https://data.census.gov/table?q=population+by+race+in+Commonwealth+of+the+Northern+Mariana+Islands&g=0400000US69$0500000)

## Motivation and Context

closes #2032 
closes #1954

## Has this been tested? How?

- python integration tests added that cover every demographic option and both geographic options
- runs on my airflow, generates expected bq tables

## Screenshots (if appropriate):

Screen Shot 2023-03-10 at 6 44 00 PM
Screen Shot 2023-03-10 at 6 45 41 PM
Screen Shot 2023-03-10 at 6 46 44 PM
Screen Shot 2023-03-10 at 6 47 25 PM
Screen Shot 2023-03-10 at 6 48 19 PM

## Types of changes

- New feature (non-breaking change which adds functionality or updates content)

## Which of our [user persona(s)](https://docs.google.com/document/d/1EASpK_THTE_uy_Yk0sut2GTVd3w5pKtKH7LpLtF-0co/) will this feature primarily benefit and how?

- Persuader: passage of legislation by presenting concise and compelling arguments (e.g. staff members, policy advisors in legislative or nonprofit settings)

Smaller territories are often neglected in public health reports and are underrepresented in Government; having robust population available allows us to confidently calculate rates for these territories and their county equivalents.

### CODE REVIEW

File: python/datasources/decia_2020_territory_population.py [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+            raw_dfs_map, breakdown, geo_level)
+        float_cols = [std_col.POPULATION_COL, std_col.POPULATION_PCT_COL]
+        column_types = gcs_to_bq_util.get_bq_column_types(
+            df, float_cols=float_cols)
+        table_name = f'by_{breakdown}_territory_{geo_level}_level'
+        gcs_to_bq_util.add_df_to_bq(df, dataset, table_name,
+                                    column_types=column_types)
+
+    def generate_breakdown_df(self,
+                              raw_dfs_map: Dict[str, pd.DataFrame],
```
Comment (eriwarr) [inline python/datasources/decia_2020_territory_population.py:156]: `dfs_by_postal_code` seems a bit more semantic to me.

File: python/datasources/decia_2020_territory_population.py [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+            # determine relevant values columns and their mapping to HET groups via tmp columns
+            if breakdown == "age":
+                value_cols = get_value_cols(AGE_CODES_TO_STD)
+                rename_map = get_rename_map(AGE_CODES_TO_STD)
+            if breakdown == "sex":
+                value_cols = get_value_cols(SEX_CODES_TO_STD)
+                rename_map = get_rename_map(SEX_CODES_TO_STD)
+            if breakdown == "race_and_ethnicity":
+                value_cols = get_value_cols(RACE_CODES_TO_STD[postal])
+                rename_map = get_rename_map(RACE_CODES_TO_STD[postal])
```
Comment (eriwarr) [inline python/datasources/decia_2020_territory_population.py:185]: Could we use constants for the breakdowns instead of the string?

File: python/ingestion/dataset_utils.py [hunk 3]
```diff
@@ -10,6 +10,76 @@
     RACE,
     UNKNOWN
 )
+from functools import reduce
+
+
+def melt_to_het_style_df(
```
Comment (eriwarr) [inline python/ingestion/dataset_utils.py:16]: I love the detail in the comments because this is an entirely new concept for me.

**Review by eriwarr — APPROVED** [review]
Just made some minor suggestions, but it looks great.🚀

File: python/datasources/vera_incarceration_county.py [hunk 4]
```diff
@@ -1,54 +1,49 @@
 import pandas as pd
 from datasources.data_source import DataSource
-from ingestion import gcs_to_bq_util
```
Comment (joshzarrabi) [inline python/datasources/vera_incarceration_county.py:3]: Why did you put this vera refactor in this PR? If its a totally separate thing I think it should be split out to a different PR.

File: python/ingestion/dataset_utils.py (hunk 3 above)
Comment (joshzarrabi) [inline python/ingestion/dataset_utils.py:16]: Very nice utility function Ben

File: python/datasources/decia_2020_territory_population.py (hunk 2 above)
Comment (joshzarrabi) [reply python/datasources/decia_2020_territory_population.py:185]: Yes, there are constants in `ingestion/constants.py`

**Review by joshzarrabi — CHANGES_REQUESTED** [review]
Great work Ben, thank you so much. 

My only comment is I think the vera stuff should be split out into its own PR.  I know that you refactored with the new melt function but I still think you should do it in another one after this.

**Comment by benhammondmusic** [comment]
> Great work Ben, thank you so much. 
> 
> 
> 
> My only comment is I think the vera stuff should be split out into its own PR.  I know that you refactored with the new melt function but I still think you should do it in another one after this.

Ya good call. I thought it was going to be a much quicker rf and ended up being a pretty big change
